projects/ngx-datatable/src/lib/components/body/body-row-wrapper.component.ts

```diff
@@ -52,7 +52,7 @@ export class DataTableRowWrapperComponent implements DoCheck, OnInit {
   @Input() detailRowHeight: any;
   @Input() row: any;
   @Input() groupedRows: any;
-  @Input() disableCheck: (row: any) => boolean;
+  @Input() disableCheck: (row: any) => boolean | BehaviorSubject<boolean>;
```

Comment:
I'm questioning the solution approach a bit: instead of fixing the actual problem, this MR *introduces* an alternative approach to set a row as `disabled`. This means that:

- The commit-message is wrong and this is rather a `feat:`.
- The API documentation in https://github.com/siemens/ngx-datatable/blob/master/docs/api/table/inputs.md shall be updated.
- The new feature should be covered at least by a minimal unit-test within https://github.com/siemens/ngx-datatable/blob/master/projects/ngx-datatable/src/lib/components/datatable.component.spec.ts.

However, I see the bug in another area: the `disableCheck` for the `DataTableRowWrapperComponent` component is executed **within** a `if (this.rowDiffer.diff(this.row)) {`, meaning only if the data changes. Moving it out of said `if` would run it during **every** `ngDoCheck()`, which already resolves the issue **without** introducing something new and on top.

@timowolf @kfenner WDYT? Do you agree with this analysis or shall we introduce an alternative API? However, this would mean that the original bug still persists when using the API the "old" way.

Reply:
@fh1ch Let's fix the bug instead of a new API.

Reply:
@fh1ch Thanks for the feedback. As you mentioned if we need to just fix the bug then we can move `disableCheck` outside of `if (this.rowDiffer.diff(this.row)) {` of `ngDoCheck()` but here I am worried about the performance aspect as `ngDoCheck` triggers on every change detections and will call `disableCheck` function for each row with every `ngDoCheck` call.

Reply:
Fixed by moving out `disableCheck` to call upon every `ngDoCheck`